src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -217,21 +217,18 @@ public function has($id)
      */
     public function get($id, $invalidBehavior = /* self::EXCEPTION_ON_INVALID_REFERENCE */ 1)
     {
-        if (isset($this->aliases[$id])) {
-            $id = $this->aliases[$id];
-        }
-
-        // Re-use shared service instance if it exists.
-        if (isset($this->services[$id])) {
-            return $this->services[$id];
-        }
-        if ('service_container' === $id) {
-            return $this;
-        }
-        if (isset($this->factories[$id])) {
-            return $this->factories[$id]();
-        }
+        return $this->services[$id]
+            ?? $this->services[$id = $this->aliases[$id] ?? $id]
```

Comment:
isn't `$this->services[$id = $this->aliases[$id] ?? $id]` enough? Do we need to check `$this->services[$id]` before?

Reply:
> this an optimization because we assume in the usual case a service is not aliased?

correct!